src/som/interpreter/nodes/dispatch/BlockDispatchNode.java

```diff
@@ -0,0 +1,64 @@
+package som.interpreter.nodes.dispatch;
+
+import com.oracle.truffle.api.CompilerDirectives.CompilationFinal;
+import com.oracle.truffle.api.Truffle;
+import com.oracle.truffle.api.dsl.Cached;
+import com.oracle.truffle.api.dsl.Fallback;
+import com.oracle.truffle.api.dsl.Specialization;
+import com.oracle.truffle.api.frame.VirtualFrame;
+import com.oracle.truffle.api.nodes.DirectCallNode;
+import com.oracle.truffle.api.nodes.IndirectCallNode;
+import com.oracle.truffle.api.nodes.Node;
+import com.oracle.truffle.api.object.DynamicObject;
+
+import som.interpreter.SArguments;
+import som.vmobjects.SBlock;
+import som.vmobjects.SInvokable;
+import som.vmobjects.SInvokable.SMethod;
+
+
+public abstract class BlockDispatchNode extends Node {
```

Comment:
Why do you need this one now?
Don't you already have a mechanisms to activate blocks?

Reply:
From the code below, I'd guess `UninitializedValuePrimDispatchNode` should do?

Reply:
`UninitializedValuePrimDispatchNode` does it. But it requires to implement the `ValuePrimitiveNode` interface so that the parent adopts the specialization in case it falls to the generic case. And this is a problem because in the Exception primitive I can not distinguish if the fallback comes from the dispatch to the block or the exception handler. From my understanding the `BlockDispatchNode` is just the same (a standard dispatch chain for blocks) but implemented with the Truffle DLS. And for free it solves that issue. Am I wrong? In case it is, my idea is to get rid of the manually written dispatch chain and finally use `BlockDispatchNode` everywhere.